hibernate-core/src/main/java/org/hibernate/dialect/LobMergeStrategy.java

```diff
@@ -18,14 +18,25 @@
  * @author Steve Ebersole
  */
 public interface LobMergeStrategy {
+
+	/**
+	 * Does this LobMergeStrategy support the merging a Lobs?
+	 *
+	 * @return true, if merging Lobs is supported; false, otherwise.
+	 */
+	default boolean supportsMerge() {
+		return true;
+	}
+
```

Comment:
To be honest I don't really understand why it's okay to consider an unfetched property as null in some cases but not in others.

From what I understand what you are trying to do is prevent the `replace` methods from returning the original value (or a deep copy of it) when the target is unfetched and the lob merge strategy is the legacy one. You want it to return `LazyPropertyInitializer.UNFETCHED_PROPERTY` instead.

My question is then: why is it okay to return the original in other cases? Does the merge happen after this call to `replace`? If so, what is the return of this method considered as: the target or the original? What is used for the other?

Another question: I suppose the behavior of calling the merge strategy on these results is built in another class. Can this class not check the fetched/unfetched state of the target and decide to call `replace` or not based on that? I really think we shouldn't be dealing with "unfetched" semantics in the type itself...

Reply:
@yrodiere, @gsmet:

`original` is a property in the `entity` being merged, (e.g, `session.merge( entity )`). `entity` may be detached or managed (i.e. already in the `PersistenceContext`).

`target` is property value in a managed entity, `managedEntity` (e.g., `managedEntity = session.merge( entity )`). The property is either a value loaded from the database, or, if lazy, it is `LazyPropertyInitializer.UNFETCHED_PROPERTY`.

Here are the possible combinations:

`original == LazyPropertyInitializer.UNFETCHED_PROPERTY`:

Resulting property in `managedEntity`: `target`
(Since `original` was uninitialized, we know that it has not been changed)

`original != LazyPropertyInitializer.UNFETCHED_PROPERTY`:

Resulting property in `managedEntity`: 

- If the property is a Lob and the dialect's `LobMergeStrategy` is `LEGACY_LOB_MERGE_STRATEGY`, then `target` should be returned (because that strategy does not support merge; iow, changes are supposed to be ignored). Replacing `target` with `null` will ultimately change the persisted value to `null`, which is not how `LEGACY_LOB_MERGE_STRATEGY` is supposed to work.

- Otherwise, a deep copy of `original` should be returned.